### PR TITLE
*: update dependency `go-sql-driver`

### PR DESCRIPTION
**Pull request by breeswish**
Cherry picked from https://github.com/pingcap/tidb/pull/4207, to support assembling DSN from structures.

@tiancaiamao @zimulala PTAL

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -193,7 +193,7 @@ imports:
   subpackages:
   - rate
 - name: google.golang.org/genproto
-  version: 09f6ed296fc66555a25fe4ce95173148778dfa85
+  version: 6b7d9516179cd47f4714cfeb0103ad1dede756c4
```
Comment (shenli) [inline glide.lock]: Why this is changed?

Comment (breeswish) [inline glide.lock]: It is changed automatically by glide.

**Comment by zimulala** [comment]
Which `go-sql-driver` version is your choice?

**Comment by breeswish** [comment]
@zimulala I used yesterday's latest commit in master

**Comment by zimulala** [comment]
LGTM

File: glide.lock (hunk 1 above)
Comment (tiancaiamao) [inline glide.lock]: Add this to glide.yaml, so next time you `make update` it won't change.

Comment (tiancaiamao) [inline glide.lock]: rest looks good to me.

**Comment by shenli** [comment]
LGTM

**Pull request merged**